src/conduits/RwaMultiSwapOutputConduit.t.sol

```diff
@@ -0,0 +1,793 @@
+        vat = vat;
+
+        spot = new Spotter(address(vat));
+        vat.rely(address(spot));
+
+        vow = new TestVow(address(vat), address(0), address(0));
+
+        usdx = new TestToken("USDX", uint8(USDX_DECIMALS));
+        usdx.mint(USDX_MINT_AMOUNT);
+
+        vat.init(ilk);
+
+        joinA = new AuthGemJoin5(address(vat), ilk, address(usdx));
+        vat.rely(address(joinA));
+
+        dai = new Dai(0);
+        daiJoin = new DaiJoin(address(vat), address(dai));
+        vat.rely(address(daiJoin));
+        dai.rely(address(daiJoin));
+
+        psm = new DssPsm(address(joinA), address(daiJoin), address(vow));
+        joinA.rely(address(psm));
+        joinA.deny(me);
+
+        pip = new DSValue();
+        pip.poke(bytes32(uint256(1 * WAD))); // Spot = $1
+
+        spot.file(ilk, bytes32("pip"), address(pip));
+        spot.file(ilk, bytes32("mat"), ray(1 * WAD));
+        spot.poke(ilk);
+
+        vat.file(ilk, "line", rad(PSM_LINE_WAD));
+        vat.file("Line", rad(PSM_LINE_WAD));
+    }
+
+    function setUp() public {
+        setUpMCDandPSM();
+
+        testUrn = vm.addr(420);
+        outputConduit = new RwaMultiSwapOutputConduit(address(dai));
+        outputConduit.file("quitTo", address(testUrn));
+        outputConduit.mate(me);
+        outputConduit.hope(me);
+        outputConduit.kiss(me);
+        outputConduit.pick(me);
+        outputConduit.clap(address(psm));
+        outputConduit.hook(address(psm));
+
+        usdx.approve(address(joinA));
+        psm.sellGem(me, USDX_MINT_AMOUNT);
+    }
+
+    function testSetWardAndEmitRelyOnDeploy() public {
+        vm.expectEmit(true, false, false, false);
+        emit Rely(address(this));
+
+        RwaMultiSwapOutputConduit c = new RwaMultiSwapOutputConduit(address(dai));
+
+        assertEq(c.wards(address(this)), 1);
+    }
+
+    function testGiveUnlimitedApprovalToPsmDaiJoinWhenClap() public {
+        RwaMultiSwapOutputConduit c = new RwaMultiSwapOutputConduit(address(dai));
+        c.clap(address(psm));
+
+        assertEq(dai.allowance(address(c), address(psm)), type(uint256).max);
+    }
+
+    function testRemoveUnlimitedApprovalFromPsmDaiJoinWhenSlap() public {
+        RwaMultiSwapOutputConduit c = new RwaMultiSwapOutputConduit(address(dai));
+        c.clap(address(psm));
+
+        assertEq(dai.allowance(address(c), address(psm)), type(uint256).max);
+
+        c.slap(address(psm));
+
+        assertEq(dai.allowance(address(c), address(psm)), 0);
+    }
+
+    function testRevertOnGemUnssuportedDecimals() public {
+        TestToken testGem = new TestToken("USDX", 19);
+        AuthGemJoin testJoin = new AuthGemJoin(address(vat), "TCOIN", address(testGem));
+        DssPsm psmT = new DssPsm(address(testJoin), address(daiJoin), address(vow));
+
+        RwaMultiSwapOutputConduit c = new RwaMultiSwapOutputConduit(address(dai));
+
+        vm.expectRevert("Math/sub-overflow");
+        c.clap(address(psmT));
+    }
```

Comment:
As I pointed in the other file, I think this problem should not be solved at this level.
If someone `clap`s an invalid PSM, they won't be able to use them anyway, so this check does not add much value IMO.